clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -416,12 +416,11 @@ const char *SYCL::Linker::constructLLVMLinkCommand(
     auto isSYCLDeviceLib = [&](const InputInfo &II) {
       const ToolChain *HostTC = C.getSingleOffloadToolChain<Action::OFK_Host>();
       StringRef LibPostfix = ".o";
-      if (HostTC->getTriple().isWindowsMSVCEnvironment() &&
-          C.getDriver().IsCLMode())
-        LibPostfix = ".obj";
-      else if (isNoRDCDeviceCodeLink(II))
+      if (isNoRDCDeviceCodeLink(II))
         LibPostfix = ".bc";
-
+      else if (HostTC->getTriple().isWindowsMSVCEnvironment() &&
+               C.getDriver().IsCLMode())
```

Comment:
I know that this was already there - but it seems redundant.  Would this be sufficient?
```suggestion
      else if (HostTC->getTriple().isWindowsMSVCEnvironment())
```

Reply:
Do we always use CL mode on windows for all possible compiler executables (icpx, dpcpp, clang, etc). I thought some of them had different effects (like even with the same oneapi install, calling icpx vs clang++ would set different flags like fastmath, and I thought CL mode might be one of them)

Reply:
CL mode is tied to `clang-cl`, which is used on Windows in general.  It also implies MSVC based targets (even when invoked on Linux).  But of course, you can target MSVC with `clang++` if you wanted, and vice-versa.  Usage of `clang++` on Windows also implies MSVC.  I think this may cause a disconnect here.  I'm fine with leaving it here and looking into an update later.

Reply:
Thanks, let me know if I should make an internal tracker for followup.